### PR TITLE
feat(playground+evals): add bedrock support

### PR DESCRIPTION
**Pull request by hassiebp**
Add bedrock support for playground + evals

----

> [!IMPORTANT]
> Add Bedrock support to playground and evals, including schema updates and API key handling.
> 
>   - **Behavior**:
>     - Add Bedrock support in `fetchLLMCompletion()` in `fetchLLMCompletion.ts` for LLM completions.
>     - Update `evalRouter` in `router.ts` to handle Bedrock-specific configurations.
>     - Modify `CreateLLMApiKeyForm.tsx` to include Bedrock-specific fields and validation.
>   - **Models**:
>     - Add `config` field to `LlmApiKeys` in `schema.prisma` and `types.ts`.
>     - Update `LLMApiKeySchema` in `types.ts` to include Bedrock configuration.
>   - **Misc**:
>     - Add `BedrockConfigSchema` and `BedrockCredentialSchema` in `customLLMProviderConfigSchemas.ts`.
>     - Remove `evalModelsByAdapter` from `types.ts`.
>     - Update `useModelParams.ts` to handle Bedrock adapter.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=langfuse%2Flangfuse&utm_source=github&utm_medium=referral)<sup> for ed376a63b6c528cc4285e9d73cc6c63f7972eb70. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: packages/shared/src/interfaces/customLLMProviderConfigSchemas.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { z } from "zod";
+
+export const BedrockConfigSchema = z.object({ region: z.string() });
+export type BedrockConfig = z.infer<typeof BedrockConfigSchema>;
+
+export const BedrockCredentialSchema = z
```
Comment (maxdeichmann) [inline packages/shared/src/interfaces/customLLMProviderConfigSchemas.ts:6]: Do we have the option to update this? I assume these values will be invalid after 1h.

File: web/src/features/llm-api-key/server/router.ts [hunk 2]
```diff
@@ -41,6 +40,7 @@ export const llmApiKeyRouter = createTRPCRouter({
             baseURL: input.baseURL,
             withDefaultModels: input.withDefaultModels,
             customModels: input.customModels,
+            config: input.config,
```
Comment (maxdeichmann) [inline web/src/features/llm-api-key/server/router.ts:43]: We are storing the secret key encrypted in the db above. In this case, the config contains secrets and is not stored encrypted.

File: web/src/features/public-api/components/CreateLLMApiKeyForm.tsx [hunk 3]
```diff
@@ -122,13 +143,30 @@ export function CreateLLMApiKeyForm({
...
+    let secretKey = values.secretKey;
+    let config: BedrockConfig | undefined;
+
+    if (currentAdapter === LLMAdapter.Bedrock) {
+      const credentials: BedrockCredential = {
+        accessKeyId: values.awsAccessKeyId ?? "",
+        secretAccessKey: values.awsSecretAccessKey ?? "",
+        sessionToken: values.awsSessionToken,
+      };
+      secretKey = JSON.stringify(credentials);
```
Comment (maxdeichmann) [inline web/src/features/public-api/components/CreateLLMApiKeyForm.tsx:155]: Good solution to the encryption!

File: packages/shared/src/interfaces/customLLMProviderConfigSchemas.ts (hunk 1 above)
Comment (hassiebp) [reply packages/shared/src/interfaces/customLLMProviderConfigSchemas.ts:6]: Only region is put into config, accesskeyId and key are stringified and encrypted as the secretkey

**Pull request merged**